logging/logging.go

```diff
@@ -873,33 +873,25 @@ func (l *Logger) writeLogEntries(entries []*logpb.LogEntry) {
 // (for example by calling SetFlags or SetPrefix).
 func (l *Logger) StandardLogger(s Severity) *log.Logger { return l.stdLoggers[s] }
 
-var reCloudTraceContext = regexp.MustCompile(`([a-f\d]+)/([a-f\d]+);o=(\d)`)
+var reCloudTraceContext = regexp.MustCompile(`([a-f\d]+)?(/([a-f\d]+))?(;o=(\d))?`)
 
 func deconstructXCloudTraceContext(s string) (traceID, spanID string, traceSampled bool) {
-	// As per the format described at https://cloud.google.com/trace/docs/troubleshooting#force-trace
+	// As per the format described at https://cloud.google.com/trace/docs/setup#force-trace
 	//    "X-Cloud-Trace-Context: TRACE_ID/SPAN_ID;o=TRACE_TRUE"
 	// for example:
 	//    "X-Cloud-Trace-Context: 105445aa7843bc8bf206b120001000/0;o=1"
 	//
 	// We expect:
-	//   * traceID:         "105445aa7843bc8bf206b120001000"
-	//   * spanID:          ""
-	//   * traceSampled:    true
-	matches := reCloudTraceContext.FindAllStringSubmatch(s, -1)
-	if len(matches) != 1 {
-		return
-	}
+	//   * traceID (optional):         "105445aa7843bc8bf206b120001000"
+	//   * spanID (optional):          "0"
+	//   * traceSampled (optional):    1 (defaults to false)
```

Comment:
Why 1? This comment was meant to illustrate the final value and implicitly traceSampled is false by default :) I'd revert the end of this comment and keep the `(optional):` or basically make it `true` instead of `1`.

Reply:
Oh yeah I was thinking about how to best represent this. Again I just took the example expected value based on our documentation instructions here: 
https://cloud.google.com/trace/docs/setup#force-trace

I'll play around with the comments to more clearly communicate what we're trying to do.